src/utils.jl

```diff
@@ -258,21 +261,46 @@ end
 """
     on_grid(func, grid)
 
-Returns an array, of the ArrayType of the device `grid` lives on, that contains the values of
-function `func` evaluated on the `grid`.
+Return an array, of the type compatible with the `device` of that the `grid` lives on,
+that contains the values of function `func` evaluated on the `grid`.
 """
-on_grid(func, grid::OneDGrid{T, A}) where {T, A} = CUDA.@allowscalar A([func(grid.x[i]) for i=1:grid.nx])
+function on_grid(func, grid::OneDGrid{T}) where T
+  f = zeros(grid.device, T, (grid.nx, ))
```

Comment:
Does `@devzeros grid.device T grid.nx f` give any performance (or otherwise) advantage? I think they do the same thing (?) so was just wondering why `zeros()` is used here vs `@devzeros` for e.g. the `Vars` function in `src/diffusion.jl`?

Reply:
with `@devzeros` you can initialize many arrays in one line

Reply:
I nuked `devzeros()` that was not required anymore. I kept `@devzeros`.